moodle/Sniffs/Commenting/CategorySniff.php

```diff
@@ -46,15 +46,20 @@ public function register() {
      * @param int $stackPtr The position in the stack.
      */
     public function process(File $phpcsFile, $stackPtr) {
-        $categoryTokens = Docblocks::getMatchingDocTags($phpcsFile, $stackPtr, '@category');
+        $docPtr = Docblocks::getDocBlockPointer($phpcsFile, $stackPtr);
```

Comment:
Uhm... haven't checked but... what should this line return if there is a file with ZERO PHPDcoc blocks?

I was adding a new extra fixture (and moving them to "Category" dir) that is a file with just one class and without any PHPDoc block (aiming to cover that "return" that is not covered) and, surprise, surprise... when I run the test, the line continues being not-covered, aka, it's returning something no matter there isn't any block.

Reply:
Or, maybe, with other words, as far as the Sniff in only looking for `T_DOC_COMMENT_OPEN_TAG`, then that line is 100% superfluous/unreachable in that case. It's impossible for the function to return null when the pointer is a `T_DOC_COMMENT_OPEN_TAG`.

So, TLDR; or we delete it, or, if we keep it as guardian for any future change in the registered tokens, we `@codeCoverageIgnore` it. I'm happy either way.

Reply:
It's intended as a safety check, so yes we could/should add the codeCoverageIgnore.

This change has no actual change - I've just broken the pointer fetch to make it more explicit.
I'll ignore the coverage there.